ecs-init/volumes/ecs_volume_driver.go

```diff
@@ -115,3 +124,9 @@ func (e *ECSVolumeDriver) Remove(req *driver.RemoveRequest) error {
 	seelog.Infof("Unmounted volume %s successfully.", req.Name)
 	return err
 }
+
+// Method to check if a volume is currently mounted.
+func (e *ECSVolumeDriver) IsMounted(volumeName string) bool {
```

Comment:
should we have a read lock on this function?

Reply:
actually I see that it's called in the Remove function which would deadlock if this tried to get a read lock

Reply:
Yeah that's a miss. I will update this method to acquire a read lock. Thanks!

Reply:
I think it wouldn't deadlock as that `Remove` method is of `AmazonECSVolumePlugin` struct and not `ECSVolumeDriver` struct.

Reply:
Updated the method to acquire read lock.